clients/src/main/java/org/apache/kafka/clients/producer/internals/BufferPool.java

```diff
@@ -228,7 +228,7 @@ public void deallocate(ByteBuffer buffer) {
     public long availableMemory() {
         lock.lock();
         try {
-            return this.availableMemory + this.free.size() * this.poolableSize;
+            return this.availableMemory + this.free.size() * (long) this.poolableSize;
```

Comment:
So I am curious. The types in this expression are `long + int + int`. Was the result being downcast to an `int`. Will the single cast to `long` fix that, or do both `int`'s need to be cast to `long`?